datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -537,6 +539,40 @@ where
     to_ticks(array, 1_000_000_000)
 }
 
+fn epoch<T>(array: &PrimitiveArray<T>) -> Result<Float64Array>
+where
+    T: ArrowTemporalType + ArrowNumericType,
+    i64: From<T::Native>,
+{
+    let mut b = Float64Builder::with_capacity(array.len());
+    match array.data_type() {
+        DataType::Timestamp(tu, _) => {
+            for i in 0..array.len() {
+                if array.is_null(i) {
+                    b.append_null();
+                } else {
+                    let scale = match tu {
```

Comment:
I double checked that this is consistent with postgres docs:  https://www.postgresql.org/docs/8.1/functions-datetime.html. 👍